google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/CloudApiManager.java

```diff
@@ -119,6 +146,175 @@ static void enableApis(Set<CloudLibrary> libraries, CloudProject cloudProject, P
     }
   }
 
+  /**
+   * Creates a new {@link ServiceAccount}, adds the supplied set of {@link Role roles} to it, and
+   * creates and downloads the service account private key to the user's file system.
+   *
+   * @param roles the set of {@link Role} to add to the new service account
+   * @param name the name of the new service account to be created
+   * @param downloadDir the {@link Path} of the download directory of the service account private
+   *     key json file
+   * @param cloudProject the current {@link CloudProject}
+   * @param project the current {@link Project}
+   */
+  static void createServiceAccountAndDownloadKey(
+      Set<Role> roles, String name, Path downloadDir, CloudProject cloudProject, Project project) {
+    Optional<CredentialedUser> user =
+        Services.getLoginService().getLoggedInUser(cloudProject.googleUsername());
+
+    if (!user.isPresent()) {
+      LOG.error("Cannot enable APIs: logged in user not found.");
+      return;
+    }
+
+    ProgressIndicator progress = ProgressManager.getInstance().getProgressIndicator();
+
+    try {
+      int numSteps = roles.isEmpty() ? 3 : 4;
+      double step = 1;
+
+      showProgress(
+          progress,
+          GctBundle.message("cloud.apis.service.account.create.account.progress.message", name),
+          step / numSteps);
+      step++;
+      ServiceAccount serviceAccount = createServiceAccount(user.get(), name, cloudProject);
+
+      if (!roles.isEmpty()) {
+        showProgress(
+            progress,
+            GctBundle.message("cloud.apis.service.account.add.roles.progress.message"),
+            step / numSteps);
+        step++;
+        addRolesToServiceAccount(user.get(), serviceAccount, roles, cloudProject);
+      }
+
+      showProgress(
+          progress,
+          GctBundle.message("cloud.apis.service.account.create.key.progress.message"),
+          step / numSteps);
+      step++;
+      ServiceAccountKey serviceAccountKey = createServiceAccountKey(user.get(), serviceAccount);
+
+      showProgress(
+          progress,
+          GctBundle.message("cloud.apis.service.account.download.key.progress.message"),
+          step / numSteps);
```

Comment:
Nit: If I get it right, here `step == numSteps` when last action is run, so it shows download message when progress bar is already full? Maybe add another step and fill progress bar after download is complete?

Reply:
what about changing the starting index to 0 instead (like it was before for enablement). that way the 1st step will show the progress as completely unstarted, and the last step will be 1 less than full. wdyt? 

For some reason, I initially wanted to change it so that the first step didn't show as unstarted but I guess it can be looked at both ways.

Reply:
I think it's good either way - adding 1 more extra step or showing first step as 0 complete, so I leave it up to you. Probably initial value of 0 is a bit easier to read.